### DISCUSSION TITLE
Board with integrated USB serial adapter, ESP32-CAM-CH340, No FTDI needed

### DISCUSSION THREAD

**Discussion by easytarget** [easytarget/esp32-cam-webserver#198] [General]
OkiDoki.

I just ordered one of the above modules (google it, I got mine from aliexpress) which is basically a AI Thinker board with a CH340 usb-serial converter onboard.
- looking at the ads I have a horrible feeling that it will still need GP0 to be held low for programming.
- But at least the FTDI board can be dispensed with, and power supply can be done with a spare USB-micro cable
- If I get this working I intend to add an affiliate link for it to the main page and make sure it gets written up in with equal importance as the AI-Thinker board.

I'll let you know... It wont arrive until the new year so I'll see what can be done with the code in the meantime

[If you want to support my work, you can try the Affiliate Link](https://s.click.aliexpress.com/e/_Ak9DAQ)
![new module](https://user-images.githubusercontent.com/4005142/146645060-5ab4ef21-b4f2-4b51-bf84-f39d4f07898b.jpg)

**Comment by easytarget** [comment]
I have the board now; it works! hurrah, and it does not need any additional hardware to program!
### No FTDI board is needed, No messing around with connecting wires while programming
- There are two buttons underneath the board next to the usb mini socket; `reset` and `flash`.
- `reset` does exactly what it says, resets and reboots the board.
- Holding the `flash` button while pressing `reset` will put the board into programming mode; after which you can simply upload from the IDE/programmer.

### Also Note:
The board came with a really simple cam server already programmed; creating a simple passwordless AP and serving a stream and images via two URI's.
  - Look at the default serial output (115200 baud) to see the camera startup log and connection details; AP name, and the URL's for the stream and image.
  - This is good since it let me test the camera before I put any of my code on it.
  - It is a very simple camera server, no control page etc, and fixed at a low resolution.

### However..
This board shows isues with high XCLK speeds; 
- Setting ```#define XCLK_FREQ_HZ 3000000;``` (3 MHz) on the last line of `myconfig.h` was needed to make the camera start and serve high resolution images without errors and visual artefacts.
- This is something that seems to be bugging a lot of recent boards.
  - my original AI-THINKER board runs with Xclk set at 20MHz, and the later boards I have needed this reduced to 16MHz, but I have seen reports that much lower speeds are needed with some recent clones.
  - The effect of this is to limit how fast the camera can send images to the module ready for sending via the network. It affects the absolute maximum frame-rate you can get.
  - The frame rate is still reasonable and acceptable for most uses; certainly good enough for a general-purpose webcam.

**Comment by jouellnyc** [comment]
I literally have this board am interested in your experience. I could not really find these online apart from ebay - https://www.ebay.com/itm/185476432256 - they say  "... press the FLASH button (always press), and then connect the module's MICRO USB port and computer's USB port through the USB cable. " no mention of RST but it seem that was the success point for you. Did your board respond / light up when you held the flash button while pressing reset?

I have tried all sorts of ways to flash this w/o success. 

I have flashed many ESP32-CAM-MB's (w/oFTDI) but this is the first  ESP32-CAM-CH340 thus far...

**Reply by 80stech** [reply]
Press and hold the flash button,, connect the USB cable, start the download and when connecting......... appears let go of the flash button.

**Reply by sonofcy** [reply]
I have a dozen or more of these boards from Aliexpress. I upload without touching any buttons. Sounds like you have fake boards.

**Comment by jouellnyc** [comment]
oh. Wow. That is completely wild / awesome at the same time. I let go just at the point when Thonny said "starting". Sure enough, it worked! Thanks for the very quick reply!

**Comment by rediculum** [comment]
For all Germans, Elektor Store offers it as well:
https://www.elektor.de/parts/development-boards/esp32-cam-ch340-development-board

**Comment by MioukiSan** [comment]
I can't understand the procedure, do I need to open the arduino IDE and then connect the esp32 cam while pressing flash and when the download is starts I'll let go the flash button?
